api4/command.go

```diff
@@ -185,6 +185,31 @@ func listCommands(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.CommandListToJson(commands)))
 }
 
+func getCommand(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireCommandId()
+	if c.Err != nil {
+		return
+	}
+
+	cmd, err := c.App.GetCommand(c.Params.CommandId)
+	if err != nil {
+		c.Err = err
+		return
+	}
```

Comment:
I think it would make more sense to fetch the command after checking the session permissions, as doing it before can cause to return a database error even when the user doesn't have permissions to manage slash commands of for the given team

Reply:
Yes, good idea.

Reply:
On a second read just saw that we need the command to check permissions for the team. Maybe `teamId` should be a requirement for the endpoint as a query param, the command list endpoint requires it as well, so it would follow the same pattern.

Another option would be to have the endpoint under the team, like `/team/{teamId}/command/{commandId}`, but it would differ too much from the rest of the endpoints, so I'm inclined to follow the query params approach [3/5](https://docs.mattermost.com/process/training.html#out-of-5).

Reply:
Ok.  I'll also have to add a check to ensure the command fetched has the same teamid as the one passed in.  If they mismatch, and user has perms for the teamid passed in, I'll return not found.

Reply:
> On a second read just saw that we need the command to get the `teamId` and check permissions. Maybe `teamId` should be a requirement for the endpoint as a query param, the command list endpoint requires it as well, so it would follow the same pattern.

Thinking about this some more...  apps like mmctl use this endpoint in their APIs to fetch a command definition before acting on it (Delete, Modify, Execute, Move).   mmctl would need to add team_id to any APIs that use GET Command.

I'm inclined to take the team_id query string back out, and figure out another way to ensure the existence of a command does not leak via error messages.  For example I could return a generic not_found 404 when the `c.App.GetCommand(c.Params.CommandId)` call fails AND when the command exists but they don't have permission.

Thoughts?

Reply:
The only problem with that approach that I can think of is that we would obscure a Server Error from a user with the right permissions if retrieving the command fails. It is not something that is going to happen often, but the possibility is there.

The alternative of requiring the `teamId` in `mmctl`'s commands is something that we are already doing in the `move` command:

```
command move newteam oldteam:command
```

But has the downside of breaking consistency with the current CLI. I'm inclined to go with this one though, if @wiersgallak thinks is alright, as it avoids any problems at the server level

Reply:
@mgdelacroix & @wiggin77 It seems odd to me to have to input a team id when you are calling a command by its explicit command ID.  My preference would be to keep the functionality we have established with the legacy.  From my understanding, database errors are rare.

Reply:
Addressed via https://github.com/mattermost/mattermost-server/pull/13598/commits/6d9a657d52b1e4291bdfdbdb14e2529cc3b91e52